MIGRATION.md

```diff
@@ -320,6 +321,10 @@ In Storybook 7 it was possible to use `addons.setConfig({...});` to configure St
 - `showPanel: boolean` is now split into `bottomPanelHeight: number` and `rightPanelWidth: number`, where the numbers represents the size of the panel in pixels.
 - `isFullscreen: boolean` is no longer supported, but can be achieved by setting a combination of the above.
 
+#### New UI and props for Button and IconButton components
+
+We used to have a lot of different buttons in `@storybook/components` that were not used anywhere. We have removed them and added a new `Button` component that can be used in all places. The `IconButton` component has also been updated to use the new `Button` component. Going forward addon creators and Storybook maintainers should use the new `Button` component. `IconButton` is still available for use, but we might remove it in the future as this is just a wrapper around the `Button` component. `Form.Button` is depreciated and will be removed in the future.
```

Comment:
We need to:

1. be specific about which exact components have been deprecated (currently it just says "a lot of different buttons") 
2. which exact props have been deprecated
3. which new props replace which old props
4. give an example of migrating from the old props to the new props

We don't need to specify stuff that we "might remove" in the future, it doesn't help the user and just creates uncertainty.

In reality this is mostly an accessible version of what's written in the source code and the types, but we can't expect users to read those at all. Consider that users will see the warning in the console with a link to this section, they'll need a way forward here to resolve the warning.

Here's a proposal:

> The X component from `@storybook/components` have been deprecated.
The following props to the `Button` component have been deprecated:
> - X use Y instead
> - X use Y instead
> - ...
> (same for `IconButton` if it has any deprecations)
> Here's an example of migrating from the old props to the new props:
> ```
> // deprecated usage
> <Button ... />
> 
> // new usage
> <Button ... />
> ```

Reply:
I just pushed an update with some new notes and clear descriptions of what props are deprecated and what is the alternative.